scripts/hud/dfjump.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const BUTTONS = {
+	ATTACK: 1 << 0,
+	JUMP: 1 << 1,
+	DUCK: 1 << 2,
+	FORWARD: 1 << 3,
+	BACK: 1 << 4,
+	USE: 1 << 5,
+	CANCEL: 1 << 6,
+	LEFT: 1 << 7,
+	RIGHT: 1 << 8,
+	MOVELEFT: 1 << 9,
+	MOVERIGHT: 1 << 10,
+	ATTACK2: 1 << 11,
+	SCORE: 1 << 16,
+	SPEED: 1 << 17,
+	WALK: 1 << 18,
+	ZOOM: 1 << 19,
+	LOOKSPIN: 1 << 25,
+	BHOPDISABLED: 1 << 29,
+	PAINT: 1 << 30,
+	STRAFE: 1 << 31
+};
+
+const COLOR_CLASS = {
+	AIR: 'dfjump__press--air',
+	GROUND: 'dfjump__press--ground'
+};
+
+class DFJump {
+	/** @type {Panel} @static */
+	static container = $('#DFJumpContainer');
+	/** @type {ProgressBar} @static */
+	static releaseBar = $('#JumpReleaseBar');
+	/** @type {ProgressBar} @static */
+	static pressBar = $('#JumpPressBar');
+	/** @type {Label} @static */
+	static releaseLabel = $('#JumpReleaseLabel');
+	/** @type {Label} @static */
+	static pressLabel = $('#JumpPressLabel');
+	/** @type {Label} @static */
+	static totalLabel = $('#JumpTotalLabel');
+
+	static onLoad() {
+		this.initializeSettings();
+		this.colorClass = COLOR_CLASS.GROUND;
+	}
+
+	static onDFJumpUpdate(releaseDelay, pressDelay, totalDelay) {
+		const releaseRatio = releaseDelay * this.inverseMaxDelay;
+		const pressRatio = Math.abs(pressDelay) * this.inverseMaxDelay;
+		const newPressColorClass = pressDelay < 0 ? COLOR_CLASS.GROUND : COLOR_CLASS.AIR;
+
+		this.releaseBar.value = releaseRatio;
+		this.pressBar.value = pressRatio;
+		this.pressBar.RemoveClass(this.colorClass);
+		this.pressBar.AddClass(newPressColorClass);
+		this.colorClass = newPressColorClass;
+
+		this.releaseLabel.text = releaseDelay.toFixed();
+		this.pressLabel.text = pressDelay.toFixed();
+		this.totalLabel.text = totalDelay.toFixed();
+	}
+
+	static setMaxDelay(newDelay) {
+		this.maxDelay = this.nullCheck(newDelay, 1);
```

Comment:
How can the delay be null?

Reply:
Idk, I didn't really dig into how the console sanitizes input. Just traumatized from all the JS errors in cgaz.js from something trying to set a panel value to `NaN px` - it makes a difference there. Since this is the denominator, I wanted to guarantee there wouldn't ever be funny business, Not a big deal, I could remove this here if it's really not needed.

Reply:
GetInt on a cvar will thankfully always return an int. Might wanna check if it's 0 though? Unless the cvar has appropriate bounds. Or might be better to check if `inverseMaxDelay` is nan

Reply:
the minimum value allowed for the max delay convar is 1. Goc's suggestion to use the ?? operator removes the need for this function, so I'll get rid of it.

I've updated the line to use a better default instead of a magic number.